### PR TITLE
mp3val: new port

### PR DESCRIPTION
**Pull request by catap**
###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on
macOS 11.5.2 20G95 x86_64
Xcode 12.5.1 12E507

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

File: audio/mp3val/Portfile [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+checksums           rmd160  25b63be18d40623324b1ce4d622bd18204965c13 \
+                    sha256  95a16efe3c352bb31d23d68ee5cb8bb8ebd9868d3dcf0d84c96864f80c31c39f \
+                    size    21406
+
+use_configure       no
+build.target        mp3val
+build.args          -f Makefile.linux
+build.env           CXX=${configure.cxx} \
+                    "CXXFLAGS=${configure.cxxflags} [get_canonical_archflags cxx]"
+
```
Comment (reneeotten) [inline audio/mp3val/Portfile:33]: this could probably be simplified by using the `makefile` PG

Comment (catap) [reply audio/mp3val/Portfile:34]: @reneeotten indeed

File: audio/mp3val/files/Makefile-cxxflags.patch [hunk 2]
```diff
@@ -0,0 +1,8 @@
+--- Makefile.linux.orig	2007-03-12 17:50:16.000000000 +0100
++++ Makefile.linux	2021-09-07 18:04:00.000000000 +0200
+@@ -1,4 +1,4 @@
+-CXXFLAGS=-Wall -O2
++CXXFLAGS := -Wall -O2 $(CPPFLAGS)
```
Comment (reneeotten) [inline audio/mp3val/files/Makefile-cxxflags.patch:5]: I don't think this is correct... why not just remove this line and just use what MacPorts sets (e.g., it should be using `-Os` and not `-O2`).

File: audio/mp3val/Portfile [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+build.target        mp3val
+build.args          -f Makefile.linux
+
+makefile.has_destdir no
+
+patchfiles          Makefile-cxxflags.patch
+
+destroot {
+    xinstall -m 755 ${worksrcpath}/mp3val ${destroot}${prefix}/bin
```
Comment (reneeotten) [inline audio/mp3val/Portfile:38]: please use `0755`

File: audio/mp3val/Portfile [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+    xinstall -d ${destroot}${prefix}/share/doc/mp3_check
+
+    foreach file {COPYING manual.html changelog.txt} {
+        xinstall -m 0644 ${worksrcpath}/${file} ${destroot}${prefix}/share/doc/mp3val
```
Comment (reneeotten) [inline audio/mp3val/Portfile:43]: this doesn't do anything as you can see when inspecting the installed files:
```
% port contents mp3val
Port mp3val contains:
  /opt/local/bin/mp3val
  /opt/local/share/doc/mp3val
```

I guess you meant to do:
```
    set docdir ${prefix}/share/doc/${name}
    xinstall -d ${destroot}${docdir}
    xinstall -m 0644 -W ${worksrcpath} COPYING manual.html changelog.txt \
        ${destroot}${docdir}
```

**Review by reneeotten — CHANGES_REQUESTED** [review]

File: audio/mp3val/Portfile [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+version             0.1.8
+revision            0
+categories          audio
+license             GPL-2
+platforms           darwin
+
+maintainers         {@catap korins.ky:kirill} openmaintainer
+
+description         mp3val is a small, high-speed, free software tool for checking MPEG audio files' integrity.
+long_description    MP3val is a small, high-speed, free software tool for checking MPEG audio files' integrity. \
```
Comment (reneeotten) [inline audio/mp3val/Portfile:16]: you can use `${description}` here and avoid repeating the first line

**Review by reneeotten — APPROVED** [review]

**Pull request merged**